### PR TITLE
Add November patch releases and mark 1.22 release as End-of-Life (EOL)

### PR DESCRIPTION
**Pull request by xmudrii**
* Add November and upcoming December patch releases to the patch releases list
* Move December patch releases for a week earlier (2022-12-07): https://groups.google.com/a/kubernetes.io/g/dev/c/GDkqm5JtiNg/m/_OHgqDgWBQAJ

/assign @saschagrunert @cpanato @puerco @Verolop 
cc @kubernetes/release-engineering

### CODE REVIEW

File: data/releases/eol.yaml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 branches:
+  - release: "1.22"
+    finalPatchRelease: "1.22.16"
+    endOfLifeDate: 2022-11-09
```
Comment (saschagrunert) [inline data/releases/eol.yaml:4]: We may have one more 1.22 in December containing infra changes.

/cc @kubernetes/k8s-infra-gcp-org-admins  @kubernetes/release-managers

Comment (xmudrii) [reply data/releases/eol.yaml:4]: @saschagrunert That's a good point. Do we want to revert this change for now and leave a note that one more 1.22 release might be issued to migrate to `registry.k8s.io`?

Comment (palnabarun) [reply data/releases/eol.yaml:4]: Should we split this into 2 PRs? One containing the December 7 patch release decision and November patch updates and another with potentially another 1.22 with the infra changes?

This way we have fresh information on the website about the early December patch releases at the least.

Comment (xmudrii) [reply data/releases/eol.yaml:4]: @palnabarun I thought about that as well. I'll take care of that.

Comment (xmudrii) [reply data/releases/eol.yaml]: I dropped changes related to 1.22 from this PR. PTAL @palnabarun @saschagrunert

**Review by jeremyrickard — APPROVED** [review]
/lgtm

File: data/releases/eol.yaml (hunk 1 above)
Comment (xmudrii) [reply data/releases/eol.yaml]: I created #37897 for 1.22.17. Let's discuss this there.

**Pull request merged**